crates/rpc/rpc-eth-api/src/helpers/fee.rs

```diff
@@ -154,11 +155,23 @@ pub trait EthFees: LoadFee {
                 base_fee_per_blob_gas.push(last_entry.next_block_blob_fee().unwrap_or_default());
             } else {
             // read the requested header range
-            let headers = LoadFee::provider(self).sealed_headers_range(start_block..=end_block).map_err(Self::Error::from_eth_err)?;
+            let mut headers = LoadFee::provider(self).sealed_headers_range(start_block..=end_block).map_err(Self::Error::from_eth_err)?;
             if headers.len() != block_count as usize {
                 return Err(EthApiError::InvalidBlockRange.into())
             }
 
+            // If the newest block is pending, we need to fetch the pending block and add it to the headers
+            let pending_block_hash = if is_pending_block {
+                if let Some(pending_block) = self.block(BlockNumberOrTag::Pending.into()).await? {
```

Comment:
we can make this slightly nicer by storing the pending block as an option and effectively collapse these two checks

as discussed also need to use 

https://github.com/paradigmxyz/reth/blob/78703b5729291f1eb198fa42c323f738ba6d608d/crates/rpc/rpc-eth-api/src/helpers/block.rs#L144-L144

because we need the receipts as well, because we need them for the rewards

this issue is somewhat related here but not a blocker

https://github.com/paradigmxyz/reth/issues/9948